install.sh

```diff
@@ -21,6 +21,15 @@ workDir=`cd ${workDir};pwd`
 
 source ${workDir}/conf/config/install_config.conf
 
+charEscape(){
+    inputParam=$1
+    if [[ $1 =~ '@' ]]; then
+        tmp=$(echo inputParam | sed "s/@/\\\\@/g")
```

Comment:
It seem we miss shell variable syntax here @tracehh 
```suggestion
        tmp=$(echo ${inputParam} | sed "s/@/\\\\@/g")
```

but the most important thing is, could we reduce the complexity of function? maybe just

```shell
function charEscape(){
  echo ${$1//@/\\\\@} 
}
```
according to https://stackoverflow.com/a/2871187/7152658

Reply:
WDYT  @tracehh

Reply:
Year, missing {} for inputParam.

Reply:
Nice advice, simplify it by reducing if

Reply:
>  Nice advice, simplify it by reducing if

Maybe you should test wether it work or not in your env.